api/address.proto

```diff
@@ -13,16 +13,25 @@ message Address {
     // matching in FilterChainMatch. For clusters, an address may be either an
     // IP or hostname to be resolved via DNS.
     google.protobuf.StringValue address = 2;
-    oneof port {
+    oneof port_specifier {
```

Comment:
I think each Resolver will be shared across many Addresses, e.g. in addition to the default DNS resolver we might have a NamedPort Resolver for Listener Addresses. So, with this arrangement we won't be able to specify the SRV service for each Address in the Resolver config.